[Lock] remove useless code

ronfroy opened the pull request:
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| License       | MIT

remove usless code

nicolas-grekas committed:
[DI] fix dumping lazy services

nicolas-grekas committed:
bug #28507 [DI] fix dumping lazy services (nicolas-grekas)

This PR was merged into the 3.4 branch.

Discussion
----------

[DI] fix dumping lazy services

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28304
| License       | MIT
| Doc PR        | -

Commits
-------

7cf1505a5b [DI] fix dumping lazy services

nicolas-grekas committed:
[DI] fix infinite loop involving self-references in decorated services

nicolas-grekas committed:
bug #28512 [DI] fix infinite loop involving self-references in decorated services (nicolas-grekas)

This PR was merged into the 3.4 branch.

Discussion
----------

[DI] fix infinite loop involving self-references in decorated services

| Q             | A
| ------------- | ---
| Branch?       | 3.4
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #28510
| License       | MIT
| Doc PR        | -

Commits
-------

20f27f9c72 [DI] fix infinite loop involving self-references in decorated services

ronfroy committed:
[Lock] remove useless code

nicolas-grekas commented on src/Symfony/Component/Lock/Store/CombinedStore.php:
```diff
@@ -149,8 +149,6 @@ public function delete(Key $key)
                 $store->delete($key);
-            } catch (\Exception $e) {
-                $this->logger->notice('One store failed to delete the "{resource}" lock.', array('resource' => $key, 'store' => $store, 'exception' => $e));
             } catch (\Throwable $e) {
```
or should we remove this one?
catching any throwable is a code smell to me, that makes debugging harder than needed.
ping @jderusse

nicolas-grekas replied:
if confirmed (it should IMHO), to be moved to 3.4.

jderusse replied:
The purpose of this "catch all" was to clean all stores even when one of theme fail with an \Error. 

The exception is silent because: a single node failure shouldn't be considered as a total failure.
This is exactly the purpose of the CombinedStore: provide a reliable virtual store allowing failure of individual store

Maybe the log level should be higher

I'm :+1:  with this PR, as the split `\Exception` and `\Throwable` was to be compatible with php 5.x which is no more required in symfony 4.x (this PR shouldn't be moved to symfony 3.4 which supports 5.5.9)

nicolas-grekas replied:
I'd like someone else to chime in here: to me, catching any "Error" should be rare:
- rethrowing
- debug
- cache-related failure

Those are the only three situations where this looks legit to me.
This is the only case in the code base where we shallow them.

ronfroy replied:
This is fixed to not catch Error anymore

nicolas-grekas replied:
Perfect timing: https://twitter.com/nikita_ppv/status/1042676176836341760

soerenbernstein committed:
[PhpUnitBridge] fix disabling DeprecationErrorHandler using phpunit.xml file

nicolas-grekas committed:
bug #28524 [PhpUnitBridge] fix disabling DeprecationErrorHandler using phpunit.xml file (soerenbernstein)

This PR was submitted for the master branch but it was squashed and merged into the 3.4 branch instead (closes #28524).

Discussion
----------

[PhpUnitBridge] fix disabling DeprecationErrorHandler using phpunit.xml file

Fixing #28519

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | don't know
| Fixed tickets | #28519
| License       | MIT

Commits
-------

3cd929bee0 [PhpUnitBridge] fix disabling DeprecationErrorHandler using phpunit.xml file

chalasr reviewed (approved):
(for 3.4)

nicolas-grekas commented:
Thank you @ronfroy.
